### ISSUE TITLE
False positive: "Unused block argument" used in nested interpolated string

### ISSUE THREAD

**Issue by kbruner** [bbatsov/rubocop#1962]
$ rubocop kitchen.rb
    Inspecting 1 file
    W

    Offenses:

    kitchen.rb:88:31: W: Unused block argument - service. If it's necessary, use _ or _service as an argument name to indicate that it won't be used.
            service_hash.each do |service, vm|
                                  ^^^^^^^

    1 file inspected, 1 offense detected

    $ rubocop -V
    0.32.0 (using Parser 2.2.2.5, running on ruby 2.2.2 x86_64-darwin14)
    $ specs gem:rubocop ruby os
    specs --version
    0.20

    gem list | grep rubocop
    rubocop (0.32.0)

    bundle --version
    Bundler version 1.10.2

    gem --version
    2.4.5

    ruby --version
    ruby 2.2.2p95 (2015-04-13 revision 50295) [x86_64-darwin14]

    system_profiler SPSoftwareDataType | grep 'System Version'
        System Software Overview:
          System Version: OS X 10.10.1 (14B25)

Code block:

        service_hash.each do |service, vm|
          execute(Instances.new(@kitchen_dir).filter_instances(vm)[0],
                  ::File.join('$HOME', 'runtime', "#{service}.start"))
        end

**Comment by kbruner** [comment]
Note also that if I wrap that block with # rubocop:disable Lint/UnusedBlockArgument I get "Unnecessary disabling of Lint/UnusedBlockArgument." instead.  :P

**Comment by jonas054** [comment]
This is very strange to me, and I can't reproduce it. What if you put only that code block in a file and run `rubocop` on it?

**Comment by kbruner** [comment]
Hmmm, yes, just with that block in a file, there's no complaint.

I've refactored the code for unrelated reasons, so I'm not actively impacted by the same issue at this point, but very odd.

**Comment by rpbaptist** [comment]
I experience the same issue in an RSpec test:

```ruby
episodes.each_with_index do |episode, i|
  Fabricate(
    :broadcast, starts_on: (i + 3).hours.from_now, episode: episode
  )
end
```

```
W: Unused block argument - episode. If it's necessary, use _ or _aflevering as an argument name to indicate that it won't be used.
      afleveringen.each_with_index do |episode, i|
                                       ^^^^^^^^^^
```

**Comment by mikegee** [comment]
I failed to duplicate on master:

```sh
rubocop master % cat test.rb
# encoding: utf-8
# frozen_string_literal: true
episodes.each_with_index do |episode, i|
  Fabricate(
    :broadcast, starts_on: (i + 3).hours.from_now, episode: episode
  )
end
rubocop master % bin/rubocop test.rb
Inspecting 1 file
.

1 file inspected, no offenses detected
```

**Comment by rpbaptist** [comment]
My apologies. Rubocop was right and I was looking at the wrong line of the file.
